### PR TITLE
judgeenv: fix problem roots not being cached

### PR DESCRIPTION
**Pull request by hieplpvip**

### CODE REVIEW

File: dmoj/judgeenv.py [hunk 1]
```diff
@@ -309,6 +296,10 @@ def get_supported_problems_and_mtimes(warnings=True):
                 else:
                     problem_dirs[problem] = problem_dir
                     problems.append((problem, os.path.getmtime(problem_dir)))
+
+    _problem_roots_cache = root_dirs
```
Comment (leduythuccs) [inline dmoj/judgeenv.py:301]: Can you remove the `root_dirs` and replace it with `_problem_roots_cache = list(root_dirs_set)`

File: dmoj/tests/test_globs.py [hunk 2]
```diff
@@ -91,10 +94,5 @@ def test_problem_roots(self):
             for problem in ex_cases:
                 self.assertRaises(KeyError, judgeenv.get_problem_root, problem)
 
-    def test_supported_problems(self):
-        with self.mock_problem_roots:
-            supported_problems = judgeenv.get_supported_problems(warnings=False)
-            self.assertEqual(list(sorted(self.supported_problems)), list(sorted(supported_problems)))
```
Comment (leduythuccs) [inline dmoj/tests/test_globs.py:97]: why don't we keep those test saparated?

**Review by leduythuccs — COMMENTED** [review]
The overall logic is looking good to me, just got some question!

File: dmoj/judgeenv.py (hunk 1 above)
Comment (hieplpvip) [reply dmoj/judgeenv.py:301]: We need to maintain the order of priority

File: dmoj/tests/test_globs.py (hunk 2 above)
Comment (hieplpvip) [reply dmoj/tests/test_globs.py:97]: `problem_roots` is not populated until `get_supported_problems` is called

File: dmoj/judgeenv.py (hunk 1 above)
Comment (leduythuccs) [reply dmoj/judgeenv.py:301]: oh yeah totally forgot about that, then could you help comment it in the code? 🤔

File: dmoj/tests/test_globs.py (hunk 2 above)
Comment (leduythuccs) [reply dmoj/tests/test_globs.py:97]: ah, okay then

**Pull request merged**